pkg/gcp/actuator/actuator.go

```diff
@@ -195,6 +210,40 @@ func (a *Actuator) sync(ctx context.Context, cr *minterv1.CredentialsRequest) er
 		return nil
 	}
 
+	stsDetected, err := utils.IsTimedTokenCluster(a.Client, ctx, logger)
+	if err != nil {
+		return err
+	}
+	if stsDetected {
+		logger.Debug("actuator detected STS enabled GCP cluster (WIF)")
+		providerSpec, err := decodeProviderSpec(minterv1.Codec, cr)
+		if err != nil {
+			return err
+		}
+		if providerSpec.ServiceAccountEmail == "" {
+			logger.Debug("CredentialsRequest has no ServiceAccountEmail, no reason to sync")
+			return nil
+		}
+		err = validateSTSProviderSpec(*providerSpec)
+		if err != nil {
+			// At least one of the fields was set indicating that the new workload identity
+			// behavior of creating the secret is desired but not all fields required were
+			// provided.
+			msg := "error validating credentials request GCP Workload Identity Federation fields"
+			return &actuatoriface.ActuatorError{
+				ErrReason: minterv1.CredentialsProvisionFailure,
+				Message:   fmt.Sprintf("%v: %v", msg, err),
+			}
+		}
+		cloudTokenPath := cr.Spec.CloudTokenPath
+		if cr.Spec.CloudTokenPath == "" {
+			logger.Debug("CredentialsRequest has no cloudTokenPath, defaulting cloudTokenPath to /var/run/secrets/kubernetes.io/serviceaccount/token")
+			cloudTokenPath = "/var/run/secrets/kubernetes.io/serviceaccount/token"
```

Comment:
Should this default to `"/var/run/secrets/openshift/serviceaccount/token"` instead? 

https://github.com/openshift/cloud-credential-operator/blob/a7f311bc118fb79d78501939d3fe392020dae1ed/pkg/apis/cloudcredential/v1/types_credentialsrequest.go#L76

https://github.com/openshift/cloud-credential-operator/blob/a7f311bc118fb79d78501939d3fe392020dae1ed/docs/gcp_workload_identity.md?plain=1#L115

Reply:
/lgtm cancel
Oh! Good catch!
https://github.com/openshift/cloud-credential-operator/blob/a68b344ed8ee1f1d27721760345781b06db2da13/pkg/cmd/provisioning/constants.go#L17

Reply:
It's the same in the AWS actuator, but I'll leave that for a follow up PR

Reply:
/lgtm

Reply:
I think there is merit to having the non-core OpenShift operators default to a more Kubernetes native path. I would be tempted to keep it as is and updater the Azure to match.